### PR TITLE
Support specifying the filepattern in `fn source` and `fn run`

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

File: cmd/config/internal/commands/source.go [hunk 1]
```diff
@@ -29,6 +29,8 @@ func GetSourceRunner(name string) *SourceRunner {
 		"output using this format.")
 	c.Flags().StringVar(&r.FunctionConfig, "function-config", "",
 		"path to function config.")
+	c.Flags().StringSliceVar(&r.InputPattern, "input-pattern", kio.DefaultMatch,
```
Comment (phanimarupaka) [inline cmd/config/internal/commands/source.go:32]: How do we signal stdin to accept json files ?

Comment (phanimarupaka) [reply cmd/config/internal/commands/source.go:32]: Tests for source ? They look similar to [this](https://github.com/kubernetes-sigs/kustomize/pull/2546/files#diff-5bd68abc0972422aef6be3cf0eb26b1aR138) I suppose? But stdin might not work.

File: cmd/config/internal/commands/run-fns.go [hunk 2]
```diff
@@ -65,6 +66,8 @@ func GetRunFnRunner(name string) *RunFnRunner {
 	r.Command.Flags().StringArrayVar(
 		&r.Mounts, "mount", []string{},
 		"a list of storage options read from the filesystem")
+	c.Flags().StringSliceVar(&r.InputPattern, "input-pattern", kio.DefaultMatch,
```
Comment (phanimarupaka) [inline cmd/config/internal/commands/run-fns.go:69]: Should we have this as a flag? I guess the behavior should be static whether to accept certain file formats or not. Other tools like terraform and cloudformation support both formats but user need not specify them through flags. Adding more such flags might make them less discoverable IMO. WDYT?

File: cmd/config/internal/commands/source.go [hunk 3]
```diff
@@ -50,7 +53,10 @@ func (r *SourceRunner) runE(c *cobra.Command, args []string) error {
 	// if there is a function-config specified, emit it
 	var functionConfig *yaml.RNode
 	if r.FunctionConfig != "" {
-		configs, err := kio.LocalPackageReader{PackagePath: r.FunctionConfig}.Read()
+		configs, err := kio.LocalPackageReader{
```
Comment (phanimarupaka) [inline cmd/config/internal/commands/source.go:56]: How would sink react to this? Does it write json files out or yaml resources to .json files? Taking json files and writing out yaml files might be confusing to user.

File: cmd/config/internal/commands/run-fns.go (hunk 2 above)
Comment (pwittrock) [reply cmd/config/internal/commands/run-fns.go:69]: I'd like to know more about why folks need to use json files before supporting them by default.  The flag gives folks the ability to opt-in, without it being on by default.  The JSON experience is not going to be as good as the YAML experience -- folks should probably just convert their JSON files to YAML files.

File: cmd/config/internal/commands/source.go (hunk 1 above)
Comment (pwittrock) [reply cmd/config/internal/commands/source.go:32]: JSON parses as YAML -- we don't need to signal anything for STDIN.

File: cmd/config/internal/commands/source.go (hunk 3 above)
Comment (pwittrock) [reply cmd/config/internal/commands/source.go:56]: Copy the PR, then write a test and find out.

File: cmd/config/internal/commands/source.go (hunk 1 above)
Comment (pwittrock) [reply cmd/config/internal/commands/source.go:32]: Yes, please add a test for source.  The intent of this PR was to give you direction, not to be the solution itself.

**Comment by pwittrock** [comment]
Closing this PR -- its goal was to be an example

**Pull request closed without merging**